common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -306,6 +306,29 @@ def non_editable_metadata_fields(self):
         non_editable_fields.extend([LibraryContentFields.mode, LibraryContentFields.source_library_version])
         return non_editable_fields
 
+    def get_tools(self):
+        """
+        Grab the library tools service or raise an error.
+        """
+        lib_tools = self.runtime.service(self, 'library_tools')
+        if not lib_tools:
+            # This error is diagnostic. The user won't see it, but it may be helpful
+            # during debugging.
+            return Response(_(u"Course does not support Library tools."), status=400)
```

Comment:
Actually, on second look, it doesn't make sense that this method sometimes returns a `LibraryTools` instance and other times a webob `Response` object. You'll need to fix that.

You could also consider making this a `@lazy` property.